### PR TITLE
fix: account linking APIs

### PR DESCRIPTION
**Pull request by sattvikc**
API spec updates for account linking

### CODE REVIEW

File: api_spec.yaml [hunk 1]
```diff
@@ -4282,7 +4711,108 @@ paths:
...
+          $ref: '#/components/responses/500'
+
+  /appid-<appId>/<tenantId>/users/by-accountinfo:
+    get:
+      tags:
+        - Core
+      operationId: getUsersByAccountInfo
+      description: |
+        Get users by email, phoneNumber or thirdParty info
+      parameters:
```
Comment (porcellus) [inline api_spec.yaml:4763]: `doUnionOfAccountInfo` seems to be missing

File: api_spec.yaml [hunk 2]
```diff
@@ -5394,9 +5938,74 @@ components:
           schema:
             $ref: '#/components/schemas/helloResponse'
 
-      
-
   schemas:
+    authRecipeUser:
```
Comment (porcellus) [inline api_spec.yaml:5942]: Please remove the old (recipe specific) user types)

**Review by porcellus — CHANGES_REQUESTED** [review]

File: api_spec.yaml [hunk 3]
```diff
@@ -22,8 +22,372 @@ tags:
...
+            application/json:
+              schema:
+                oneOf:
+                  - type: object
+                    properties:
+                      status:
+                        $ref: '#/components/schemas/statusOK'
+                      wasAlreadyAPrimaryUser:
+                        type: boolean
+                        example: true
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply api_spec.yaml:212]: Added

File: api_spec.yaml [hunk 4]
```diff
@@ -22,8 +22,372 @@ tags:
...
+                    properties:
+                      status:
+                        type: string
+                        enum:
+                          [
+                            'RECIPE_USER_ID_ALREADY_LINKED_WITH_PRIMARY_USER_ID_ERROR',
+                          ]
+                      primaryUserId:
+                        type: string
+                        example: fa7a0841-b533-4478-9253-0fde890c576
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply api_spec.yaml:236]: it's not implemented that way in the core, does core need updating?

File: api_spec.yaml [hunk 5]
```diff
@@ -22,8 +22,372 @@ tags:
...
+            application/json:
+              schema:
+                oneOf:
+                  - type: object
+                    properties:
+                      status:
+                        $ref: '#/components/schemas/statusOK'
+                      accountsAlreadyLinked:
+                        type: boolean
+                        example: true
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply api_spec.yaml:293]: Added

File: api_spec.yaml [hunk 6]
```diff
@@ -1127,11 +1494,20 @@ paths:
           description: update user's information
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply api_spec.yaml:1494]: Done

File: api_spec.yaml [hunk 7]
```diff
@@ -1283,7 +1664,7 @@ paths:
                       status:
                         $ref: '#/components/schemas/statusOK'
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply api_spec.yaml:1665]: yup, done

File: api_spec.yaml (hunk 1 above)
Comment (sattvikc) [reply api_spec.yaml:4763]: Done

File: api_spec.yaml (hunk 2 above)
Comment (sattvikc) [reply api_spec.yaml:5955]: Done

File: api_spec.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (porcellus) [reply api_spec.yaml:238]: No, it's fine.

**Review by porcellus — APPROVED** [review]
LGTM (comparing w/ node)

File: api_spec.yaml [hunk 8]
```diff
@@ -22,8 +22,378 @@ tags:
   - name: Core
   - name: Dashboard Recipe
   - name: Totp Recipe
+  - name: Account Linking Recipe
 
 paths:
```
Comment (rishabhpoddar) [inline api_spec.yaml:27]: mark apis that are deprecated

File: api_spec.yaml [hunk 9]
```diff
@@ -5539,44 +6110,49 @@ components:
...
-    
+
     userMetadata:
       type: object
       description: should be a JSON object (not a JSON literal nor an array)
-      example: { "preferences": { "theme":"dark" }, "notifications": { "email": true }, "todos": ["example"] }
+      example:
+        {
+          'preferences': { 'theme': 'dark' },
+          'notifications': { 'email': true },
```
Comment (rishabhpoddar) [inline api_spec.yaml:6125]: change example CDI to 4.0

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Pull request merged**